source/agora/common/EnrollmentManager.d

```diff
@@ -205,6 +205,78 @@ public class EnrollmentManager
         this.db.execute("DELETE FROM validator_set WHERE key = ?", enroll_hash[]);
     }
 
+    /***************************************************************************
+
+        In validatorSet DB, return the enrolled block height.
+
+        Params:
+            enroll_hash = key for an enrollment block height
+
+        Returns:
+            the enrolled block height.
+            Return size_t if no matching key exists.
+
+    ***************************************************************************/
+
+    public size_t getEnrolledHeight (const ref Hash enroll_hash) @trusted
+    {
+        try
+        {
+            auto results = this.db.execute("SELECT enrolled_height FROM validator_set" ~
+                " WHERE key = ?", enroll_hash[]);
+            if (!results.empty)
+                return results.oneValue!(size_t);
+            else
+                return size_t.init;
+        }
+        catch (Exception ex)
+        {
+            log.error("Database operation error {}", ex);
+            return size_t.init;
+        }
+    }
+
+    /***************************************************************************
+
+        Update the enrolled height of the validatorSet DB.
+
+        Params:
+            enroll_hash = enrollment blockheight to update enroll hash
+            block_height = enrolled blockheight
+
+        Returns:
+            Update execution status
```

Comment:
Use void?

Reply:
applied the suggestion. `true if the updated blockheight`

Reply:
If you use `void` you'll have to `throw`.
I mean that "Update execution status" is not descriptive. If someone read it in 1 year, they won't have a clue what it is.

Something like "`true` if the update operation was successful, `false` otherwise" is a bit better.

Reply:
applied the suggestion.